staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -293,7 +293,20 @@ func (c *dynamicResourceClient) Get(ctx context.Context, name string, opts metav
 	if err != nil {
 		return nil, err
 	}
-	return uncastObj.(*unstructured.Unstructured), nil
+
+	obj, ok := uncastObj.(*unstructured.Unstructured)
+	if ok {
+		return obj, nil
+	}
+
+	// If "items" exists in the response, it will become a list
```

Comment:
When this can happen? do you have "real" reproducer?
I'd like to see a repro with an integration test first test/integration/client/dynamic_client_test.go

Reply:
@ryantxu 
do you have an extension apiserver?

Reply:
This is in a custom apiserver -- we added a [subresource](https://github.com/grafana/grafana/blob/main/pkg/registry/apis/identity/team/rest_members.go#L64) that returns a list, and it panics when using dynamic client.  (NOTE the server is still very much a work-in-progress)

Things are fine if the rest.Connector does not return a field with "items" in the name because the dynamic client blindly casts.  

sorry for the slow reply!  cc @aojea @xuzhenglun